src/waveform/widgets/allshader/hsvwaveformwidget.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "util/class.h"
+#include "waveform/widgets/allshader/waveformwidget.h"
+
+class WaveformWidgetFactory;
+
+namespace allshader {
+class HSVWaveformWidget;
+}
+
+class allshader::HSVWaveformWidget final : public allshader::WaveformWidget {
+    Q_OBJECT
+  public:
+    ~HSVWaveformWidget() override;
```

Comment:
no destructor necessary as far as I can tell because the WaveformWidget one is already non-virtual, so its not called polymorphically, right?
```suggestion
```

Reply:
Removed